src/transaction/log_manager.c

```diff
@@ -3380,6 +3395,11 @@ log_append_savepoint (THREAD_ENTRY * thread_p, const char *savept_name)
       return NULL;
     }
 
+  if (prm_get_bool_value (PRM_ID_SUPPLEMENTAL_LOG) == true)
+    {
+      log_append_supplement_user_at_start (thread_p, tdes, LOG_PRIOR_LSA_WITHOUT_LOCK);
```

Comment:
Could you let me know why is it needed in `log_append_savepoint ()`? If this is because you want to catch all the points `head_lsa` is initialized, it seems that you need to put it into all other `append_log_...()` function like log_append_ha_server_state(), log_append_empty_record(), log_append_postpone(), and so on.

Reply:
Technically, all `log_append_..()` functions that eventually call `prior_lsa_next_record_internal()`.

Reply:
DDL statements like 'CREATE', and 'ALTER' tables, indexes calls log_append_savepoint() at first. 
( ex) do_alter() -> tran-system_savepoint() -> log_append_savepoint() )
Other DML statements calls log_append_undo/redo/undoredo_crumbs. 
I just want to catch the moment that transaction starts with DDL or DML.

Reply:
What if a HA status log is appended before starting DDL or DML? It seems to be unsafe to assume all DDL will do Savepoint before starting DDL operation. Savepoint is just one of the interfaces given to a user by SM, and It might not be used. We should assume some operations don't use savepoint although now all use it. The way I see it, you're making a restriction all the DDL has to use Savepoint.

Reply:
What if a user do other jobs that doesn't call log_append_savepoint() or `log_append_undo/redo..()` and then do DDL or DML operation in a transaction?

Reply:
1. OK
2. No, the USER I said is not a database user, but all clients that can call SM interface like CAS.

Reply:
Plus, even though all DDLs use Savepoint when starting DDL, I think it's just clients (CAS, csql and so on) use them together now and SM should assume they are called independently.

Reply:
OK, I understand what you mean. I also need to consider that one. thanks

Reply:
I think that this approach(check it every log_append_* function call) is needed to reconsider. Please, find a better place than now.